src/JSONValueNode.js

```diff
@@ -4,41 +4,57 @@ import React, { PropTypes } from 'react';
  * Renders simple values (eg. strings, numbers, booleans, etc)
  */
 
-const JSONValueNode = ({
-  nodeType,
-  styling,
-  labelRenderer,
-  keyPath,
-  valueRenderer,
-  value,
-  valueGetter
-}) => (
-  <li
-    {...styling('value', nodeType, keyPath)}
-  >
-    <label {...styling(['label', 'valueLabel'], nodeType, keyPath)}>
-      {labelRenderer(keyPath, nodeType, false, false)}
-    </label>
-    <span {...styling('valueText', nodeType, keyPath)}>
-      {valueRenderer(valueGetter(value), value, ...keyPath)}
-    </span>
-  </li>
-);
+export default class JSONValueNode extends React.Component {
+  static propTypes = {
+    nodeType: PropTypes.string.isRequired,
+    styling: PropTypes.func.isRequired,
+    labelRenderer: PropTypes.func.isRequired,
+    keyPath: PropTypes.arrayOf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+    ).isRequired,
+    valueRenderer: PropTypes.func.isRequired,
+    value: PropTypes.any,
+    valueGetter: PropTypes.func
+  };
 
-JSONValueNode.propTypes = {
-  nodeType: PropTypes.string.isRequired,
-  styling: PropTypes.func.isRequired,
-  labelRenderer: PropTypes.func.isRequired,
-  keyPath: PropTypes.arrayOf(
-    PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-  ).isRequired,
-  valueRenderer: PropTypes.func.isRequired,
-  value: PropTypes.any,
-  valueGetter: PropTypes.func
-};
+  static defaultProps = {
+    valueGetter: value => value
+  };
 
-JSONValueNode.defaultProps = {
-  valueGetter: value => value
-};
+  constructor() {
+    super();
+    this.state = { hover: false };
+    this.handleMouseOver = this.handleMouseOver.bind(this);
+    this.handleMouseOut = this.handleMouseOut.bind(this);
```

Comment:
Same here (so there's no need for constructor, just define `state` as property, as well)